src/command_modules/azure-cli-network/tests/test_network_commands.py

```diff
@@ -452,26 +452,28 @@ def body(self):  # pylint: disable=too-many-statements
         self.cmd('network lb rule list {}'.format(lb_rg), checks=JMESPathCheck('length(@)', 0))
 
 
-class NetworkLocalGatewayScenarioTest(ResourceGroupVCRTestBase):
-    def __init__(self, test_method):
-        super(NetworkLocalGatewayScenarioTest, self).__init__(__file__, test_method, resource_group='local_gateway_scenario')
-        self.name = 'cli-test-loc-gateway'
-        self.resource_type = 'Microsoft.Network/localNetworkGateways'
-
-    def test_network_local_gateway(self):
-        self.execute()
+class NetworkLocalGatewayScenarioTest(ScenarioTest):
 
-    def body(self):
-        self.cmd('network local-gateway create --resource-group {} --name {} --gateway-ip-address 10.1.1.1'.format(self.resource_group, self.name))
-        self.cmd('network local-gateway list --resource-group {}'.format(self.resource_group), checks=[JMESPathCheck('type(@)', 'array'), JMESPathCheck("length([?type == '{}']) == length(@)".format(self.resource_type), True), JMESPathCheck("length([?resourceGroup == '{}']) == length(@)".format(self.resource_group), True)])
-        self.cmd('network local-gateway show --resource-group {} --name {}'.format(self.resource_group, self.name), checks=[JMESPathCheck('type(@)', 'object'), JMESPathCheck('type', self.resource_type), JMESPathCheck('resourceGroup', self.resource_group), JMESPathCheck('name', self.name)])
-        try:
-            self.cmd('network local-gateway delete --resource-group {} --name {}'.format(self.resource_group, self.name))
-            # Expecting no results as we just deleted the only local gateway in the resource group
-            self.cmd('network local-gateway list --resource-group {}'.format(self.resource_group), checks=NoneCheck())
-        except CLIError:
-            # TODO: Remove this once https://github.com/Azure/azure-cli/issues/2373 is fixed
-            pass
+    @ResourceGroupPreparer(name_prefix='local_gateway_scenario')
```

Comment:
The created resource groups should all have the relevant test tag so that would be how I'd recommend batch deleting them. I like the human readable prefix so I know what test is running. I can examine the resource group if need be, which is much much harder to do when the name is "clitest.jibberish".